src/utils/ParseRouteUtils.js

```diff
@@ -532,8 +534,137 @@ function parseRoute(resp: Object, destinationName: string) {
     }));
 }
 
+/**
+ * Return a more detailed route information returned from parseRoute function,
+ * including travelDistance, totalDuration, and routeSummary.
+ *
+ * Example return object:
+ [
+ {  departureTime: '2018-10-22T03:44:19Z',
+    arrivalTime: '2018-10-22T04:01:46Z',
+    directions: [ [Object], [Object], [Object], [Object] ],
+    startCoords: { lat: 42.441603931224435, long: -76.48638788207742 },
+    endCoords: { lat: 42.45662677611252, long: -76.47693624444763 },
+    startName: 'Risley Hall - Shelter',
+    endName: 'Ithaca Commons - Seneca St',
+    boundingBox:
+     { minLat: 42.441596,
+       minLong: -76.490387,
+       maxLat: 42.456818,
+       maxLong: -76.471642 },
+    travalDistance: 1231,
+    totalDuration: 12,
+    numberOfTransfers: 1,
+    routeSummary: [...],
+  },
+ ...
+  ]
+ * @param resp
+ * @param destinationName
+ * @returns {Promise<Array>}
+ */
+async function parseDetailedRoute(resp: Object, destinationName: string) {
+    const routeInfo = await parseRoute(resp, destinationName);
+
+    const {
+        departureTime,
+        arrivalTime,
+        directions,
+        startCoords,
+        endCoords,
+        boundingBox,
+        numberOfTransfers,
+    } = routeInfo;
+
+    let travelDistance = geolib.getDistance(
+        { latitude: startCoords.lat, longitude: startCoords.long },
+        { latitude: endCoords.lat, longitude: endCoords.long }
+    );
+
+    travelDistance *= METERS_TO_MILES;
+    routeInfo.travelDistance = travelDistance;
+
+    let totalDuration = new Date(arrivalTime) - new Date(departureTime);
+    totalDuration /= 1000;
+    totalDuration /= 60;
+    routeInfo.totalDuration = Math.round(totalDuration);
+
+    routeInfo.startName = routeInfo.directions[0].name;
+    routeInfo.endName = destinationName;
+
+    // routeSummary
+
+    /* routeSummaryElement {
+        stopName: String,
+        direction: RouteSummaryDirection?,
+        shouldStayOnBus: Bool,
+    }
+    */
+
+    /* routeSummaryDirection {
+        type: null || 'walk' || 'bus',
+        routeNumber: null || Int,
+    }
+    */
+
+    let routeSummary = [];
+
+    let walkOnlyRoute = true;
+    for (let i = 0; i < directions.length; i++) {
```

Comment:
You don't need to iterate through the directions here. In a walking route the route number and stops fields are null or empty which indicates walking only.